legion/engine/core/ecs/registry.cpp

```diff
@@ -162,6 +162,13 @@ namespace legion::core::ecs
     void Registry::destroyEntity(entity target, bool recurse)
     {
         OPTICK_EVENT();
+
+        // Destroy every component in the composition and clear the composition.
+        auto& composition = instance.m_entityCompositions.at(target->id);
+        for (auto& componentId : composition)
+            getFamily(componentId)->destroy_component(target);
+        composition.clear();
+
```

Comment:
by doing this before the mark entity destruction in the filters, this introduces a race condition with filters on multiple threads.

Reply:
I'll have a better look at this later when my internet returns, KPN is out in my neighbourhood